src/settings/components/SettingsMain.tsx

```diff
@@ -1,70 +1,29 @@
-import CopilotPlugin from "@/main";
 import React from "react";
-import { useSettingsContext } from "../contexts/SettingsContext";
 import AdvancedSettings from "./AdvancedSettings";
 import ApiSettings from "./ApiSettings";
 import CopilotPlusSettings from "./CopilotPlusSettings";
 import GeneralSettings from "./GeneralSettings";
 import QASettings from "./QASettings";
+import { resetSettings } from "@/settings/model";
+import { ResetSettingsConfirmModal } from "@/components/modals/ResetSettingsConfirmModal";
 
-const SettingsMain: React.FC<{ plugin: CopilotPlugin }> = ({ plugin }) => {
-  const { settings, updateSettings, saveSettings, resetSettings } = useSettingsContext();
-
+const SettingsMain: React.FC = () => {
   return (
-    <>
-      <h2>Copilot Settings</h2>
-      <div className="button-container">
-        <button className="mod-cta" onClick={saveSettings}>
-          Save and Reload
-        </button>
-        <button className="mod-cta" onClick={resetSettings}>
+    <div style={{ display: "flex", flexDirection: "column", gap: "1rem" }}>
+      <h1 style={{ display: "flex", alignItems: "center", justifyContent: "space-between" }}>
+        Copilot Settings
+        <button onClick={() => new ResetSettingsConfirmModal(app, () => resetSettings()).open()}>
```

Comment:
added a modal to confirm the change here. This button becomes more powerful than before and it can get annoying if people accidentally click on it and lose all of their keys